src/utils/app.ts

```diff
@@ -0,0 +1,62 @@
+import { CodeMaker } from "codemaker";
+import kebabCase from "lodash.kebabcase";
+import type { Imports } from "./imports";
+
+export interface StackProps {
+  name: string;
+}
+
+export interface AppTemplate {
+  name: string;
+  stacks: StackProps[];
+}
+
+interface AppBuilderProps {
+  imports: Imports;
+  appName: string;
+  stacks: StackProps[];
+  outputFile: string;
+  outputDir: string;
+  comment?: string;
+}
+
+export class AppBuilder {
+  config: AppBuilderProps;
+
+  code: CodeMaker;
+
+  constructor(props: AppBuilderProps) {
+    this.config = props;
+
+    this.code = new CodeMaker({ indentationLevel: 2 });
+    this.code.closeBlockFormatter = (s?: string): string => s ?? "}";
+  }
+
+  async constructCdkFile(): Promise<void> {
+    this.code.openFile(this.config.outputFile);
+    if (this.config.comment) {
+      this.code.line(this.config.comment);
+      this.code.line();
+    }
+
+    this.config.imports.render(this.code);
+
+    this.code.line("const app = new App();");
+
+    this.config.stacks.forEach((stack) => {
+      this.code.line(
+        `new ${stack.name}(app, "${stack.name}", { app: "${kebabCase(
```

Comment:
I think we may need to convert `stack.name` to upper camel case here?

Reply:
Ah yes. From what I remember, we make the assumption that the stack name provided is already in upper camel case but it would be better to convert it. I was already tempted to include both the upper camel case and kebab case values in the config object to avoid repeatedly converting so maybe that change is worth making. Do you think it's better to make that change in this PR or in a follow up?

Reply:
Perhaps something for a follow-up PR; I just happened to test this PR out using an edge case stack name, so thought I'd mention it!